mkosi/__init__.py

```diff
@@ -6284,7 +6766,11 @@ def setup_ssh(
     # which introduces non-trivial issue when trying to cache it.
 
     if not cached:
-        run(["systemctl", "--root", root, "enable", unit])
+        if args.distribution == Distribution.gentoo:
+            cmdline = ["systemctl", "enable", unit]
+            run_workspace_command(args, root, cmdline)
+        else:
+            run(["systemctl", "--root", root, "enable", unit])
```

Comment:
Is this for when the host system is not using systemd as PID1? If we want to support that, it should work on all distros, not just one. Let's not do this here and make it a separate issue/PR for whether and how we want to support this

Reply:
not really, problem _seems_ to be coming from [this](https://github.com/systemd/systemd/blob/aca8ecc39892dda22a1cc0ea6860cbc0774897f3/src/systemctl/systemctl-sysv-compat.c#L167) [src/systemctl/systemctl-sysv-compat.c#L167]

which ends up with [this](https://github.com/257/mkosi/runs/2994079593?check_suite_focus=true#step:9:615):
```
Synchronizing state of sshd.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install --root=/home/runner/work/mkosi/mkosi/mkosi.output/.mkosi-3koft3i8/root enable sshd
chroot: failed to run command ‘/usr/sbin/update-rc.d’: No such file or directory
```

unlink'ing ``/etc/init.d/sshd`` seems to *fix* it; a hack of course but i don't have a better solution; *fixing* this upstream (systemd) may not even make sense...

Reply:
> unlink'ing `/etc/init.d/sshd` seems to _fix_ it; a hack of course but i don't have a better solution; _fixing_ this upstream (systemd) may not even make sense...

on second thought, i rather not remove what package manager installed on the system.  using ``run_workspace_command()`` is more sensible, me thinks. your thoughts?

Reply:
going with the `os.unlink()`.